Libraries/LibMarkdown/Heading.h

```diff
@@ -35,15 +36,20 @@ namespace Markdown {
 
 class Heading final : public Block {
 public:
-    virtual ~Heading() override {}
+    Heading(Text&& m_text, int m_level)
+        : m_text(move(m_text))
+        , m_level(m_level)
+    {
+    }
+    virtual ~Heading() override { }
 
     virtual String render_to_html() const override;
     virtual String render_for_terminal() const override;
-    virtual bool parse(Vector<StringView>::ConstIterator& lines) override;
+    static OwnPtr<Heading> parse(Vector<StringView>::ConstIterator& lines);
 
 private:
     Text m_text;
-    int m_level { -1 };
+    int m_level;
```

Comment:
It still needs done initial value even if it's directly assigned in the constructor, because @awesomekling is paranoid for that sort of thing. When you make it a `size_t`, you wouldn't be able to use -1, so we'll have to do with 0.